container_linux.go

```diff
@@ -135,6 +135,7 @@ func (c *linuxContainer) commandTemplate(p *Process, childPipe *os.File) (*exec.
 	cmd.Stdin = p.Stdin
 	cmd.Stdout = p.Stdout
 	cmd.Stderr = p.Stderr
+	cmd.ExtraFiles = p.ExtraFiles
```

Comment:
Ah, good point, we want the `childPipe` to get fd3 so it needs to go first.  In go you can do this:

`cmd.ExtraFiles = append([]*os.File{childPipe}, p.ExtraFiles...)` to do it all in one line